src/Features/Core/Portable/Completion/CompletionServiceOptions.cs

```diff
@@ -21,5 +19,12 @@ internal static class CompletionServiceOptions
         /// </summary>
         public static readonly Option2<bool> DisallowAddingImports
             = new(nameof(CompletionServiceOptions), nameof(DisallowAddingImports), defaultValue: false);
+
+        /// <summary>
+        /// Timeout value used for time-boxing completion of unimported extension methods.
+        /// Value less than 0 means no timebox; value == 0 means immediate timeout (for testing purpose)
+        /// </summary>
+        public static readonly Option2<int> TimeoutInMillisecondsForExtensionMethodImportCompletion
+            = new(nameof(CompletionServiceOptions), nameof(TimeoutInMillisecondsForExtensionMethodImportCompletion), defaultValue: 500);
```

Comment:
is there some other value we can read in for this?  i.e. i feel like this should be driven somewhat by:

![image](https://user-images.githubusercontent.com/4564579/96796686-959bf880-13b4-11eb-85a4-72a3c5056f55.png)

Reply:
Yea, we are actually discussing this right now. Will probably come up with something soon (not in this PR though)

Reply:
ok thanks :)

Reply:
I wonder what's the designed user scenario here?
After this change, if my computer is slow and it can't get the completion finished in 0.5s then does it mean I am unable to use this feature?

Reply:
Right now, we unconditionally apply the timebox, so it means the if it takes more than 500ms, unimported extension methods wouldn't show by default even if you enabled the feature. But when that happens, you can still use expander button to make explicit request, which ignores the timebox.